src/modules/notifications/selectors/notification-state.ts

```diff
@@ -0,0 +1,46 @@
+import { createSelector } from "reselect";
+import { selectMarketsDataState } from "src/select-state";
+import { selectMarket } from "modules/markets/selectors/market";
+import { sortOpenOrders } from "modules/orders/selectors/open-orders";
+import { constants } from "services/constants";
+
+
+export const selectResolvedMarketsOpenOrders = createSelector(
+  selectMarketsDataState,
+  (marketData: { [name: string]: any }) => {
+    const marketIds = Object.keys(marketData);
+
+    if (marketIds.length > 0) {
+      // Get all the users active markets and filter on FINALIZED
+      const finalizedMarkets = marketIds
+        .filter(market => marketData[market].reportingState === constants.REPORTING_STATE.FINALIZED)
+        .map(marketId => {
+          return {
+            marketId,
+            marketName: marketData[marketId].description
+          }
+        });
+
+      // Get all Open Orders for a provided market
+      const hasOpenOrders = (marketId: string)  => {
+        const market = selectMarket(marketId);
```

Comment:
Instead of going though the markets via marketsData.

You can use `selectMarket` and filter on marketStatus === 'resolved' this will get `finalized` and `awaiting_finalization` markets.

Also don't use `userOpenOrders` on outcomes, that is going away. user `market.userOpenOrders` it'll be faster.

also using these two steps you don't have to filter the markets in the end they are already filtered.

could do something like this, as an example. prob easier with a .map 🤷‍♂️ 
```
  const userOrders = selectMarkets(state).reduce(
    (p, market) => [...p, ...market.userOpenOrders],
    []
  );
```

Reply:
👍 💯